src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3030,14 +3036,18 @@ jint Arguments::finalize_vm_init_args(bool patch_mod_javabase) {
 
 #if INCLUDE_CDS
   if (DumpSharedSpaces) {
-    // Compiler threads may concurrently update the class metadata (such as method entries), so it's
-    // unsafe with -Xshare:dump (which modifies the class metadata in place). Let's disable
-    // compiler just to be safe.
-    //
-    // Note: this is not a concern for dynamically dumping shared spaces, which makes a copy of the
-    // class metadata instead of modifying them in place. The copy is inaccessible to the compiler.
-    // TODO: revisit the following for the static archive case.
-    set_mode_flags(_int);
+    if (!mode_flag_cmd_line) {
+      // By default, -Xshare:dump runs in interpreter-only mode, which is required for deterministic archive.
+      //
+      // If your classlist is large and you don't care about deterministic dumping, you can use
+      // -Xshare:dump -Xmixed to improve dumping speed.
+      set_mode_flags(_int);
+    } else if (_mode == _comp) {
+      // -Xcomp may use excessive CPU for the test tiers. Also, -Xshare:dump runs a small and fixed set of
+      // Java code, so there's not much benefit in running -Xcomp.
+      log_info(cds)("reduced -Xcomp to -Xmixed");
```

Comment:
Is it preferable to reduce the user-selected option over throwing an error for incompatible options?

Reply:
We have many test jobs that add `-Xcomp` to every test case, so throwing an error will cause these tests to fail.

If we don't reduce the strength to `-Xmixed`, then many of these tests will fail due to timeout, forcing us to change the timeout value in these tests.

So I think changing to `-Xmixed` is a good compromise to do some testing with the JIT during `-Xshare:dump` without being intrusive.

Reply:
That makes perfect sense! I am still concerned about the idea taking a user option and reducing it to an option they didn't ask for... I will trust your judgement